exporter/awsxrayexporter/awsxray_test.go

```diff
@@ -42,6 +45,23 @@ func TestTraceExport(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+// test when 2 span have invalid xray id and 2 span have valid xray id
+func TestMixedValidSpanTraceExport(t *testing.T) {
+	traceExporter := initializeTracesExporter()
+	ctx := context.Background()
+	td := constructMixedSpanData()
+	err := traceExporter.ConsumeTraces(ctx, td)
+	assert.NotNil(t, err)
+	err = traceExporter.Shutdown(ctx)
+	assert.Nil(t, err)
+}
+
+func TestMixedValidSpanTraceResourceExtraction(t *testing.T) {
+	td := constructMixedSpanData()
+	logger, _ := zap.NewProduction()
+	assert.Len(t, extractResourceSpans(generateConfig(), logger, td), 2, "Only 2 spans have a valid xray trace id")
```

Comment:
It would have been nice to not need to expose this function for the test, but I can see you are checking the length and I think that's useful. I can't think of any other way to do it so this makes sense to me.

Reply:
Yeah I was thinking the same thing. I think this is fine since this is still a private method.